src/python_testing/TC_EEVSE_2_2.py

```diff
@@ -0,0 +1,319 @@
+        circuit_capacity = await self.read_evse_attribute_expect_success(attribute="CircuitCapacity")
+        expected_max_charge = min(max_charge_current, circuit_capacity)
+        await self.check_evse_attribute("MaximumChargeCurrent", expected_max_charge)
+
+        self.validate_energy_transfer_started_event(event_data, session_id, expected_state, expected_max_charge)
+
+        self.step("7")
+        # Sleep for the charging duration plus a couple of seconds to check it has stopped
+        time.sleep(charging_duration + 2)
+        # check EnergyTransferredStoped (EvseStopped)
+        event_data = events_callback.WaitForEventReport(Clusters.EnergyEvse.Events.EnergyTransferStopped)
+        expected_reason = Clusters.EnergyEvse.Enums.EnergyTransferStoppedReasonEnum.kEVSEStopped
+        self.validate_energy_transfer_stopped_event(event_data, session_id, expected_state, expected_reason)
+
+        self.step("7a")
+        await self.check_evse_attribute("State", Clusters.EnergyEvse.Enums.StateEnum.kPluggedInDemand)
+
+        self.step("7b")
+        await self.check_evse_attribute("SupplyState", Clusters.EnergyEvse.Enums.SupplyStateEnum.kDisabled)
+
+        self.step("8")
+        charge_until = NullValue
+        min_charge_current = 6000
+        max_charge_current = 12000
+
+        await self.send_enable_charge_command(charge_until=charge_until, min_charge=min_charge_current, max_charge=max_charge_current)
+        event_data = events_callback.WaitForEventReport(Clusters.EnergyEvse.Events.EnergyTransferStarted)
+
+        self.step("8a")
+        await self.check_evse_attribute("State", Clusters.EnergyEvse.Enums.StateEnum.kPluggedInCharging)
+
+        self.step("8b")
+        await self.check_evse_attribute("SupplyState", Clusters.EnergyEvse.Enums.SupplyStateEnum.kChargingEnabled)
+
+        self.step("8c")
+        await self.check_evse_attribute("ChargingEnabledUntil", charge_until)
+
+        self.step("8d")
+        await self.check_evse_attribute("MinimumChargeCurrent", min_charge_current)
+
+        self.step("8e")
+        circuit_capacity = await self.read_evse_attribute_expect_success(attribute="CircuitCapacity")
+        expected_max_charge = min(max_charge_current, circuit_capacity)
+        await self.check_evse_attribute("MaximumChargeCurrent", expected_max_charge)
+
+        # from step 8 above - validate event
+        self.validate_energy_transfer_started_event(event_data, session_id, expected_state, expected_max_charge)
+
+        self.step("9")
+        # This will only work if the optional UserMaximumChargeCurrent attribute is supported
+        supported_attributes = await self.get_supported_energy_evse_attributes()
+        if Clusters.EnergyEvse.Attributes.UserMaximumChargeCurrent.attribute_id in supported_attributes:
+            logging.info("UserMaximumChargeCurrent is supported...")
+            user_max_charge_current = 6000
+            await self.write_user_max_charge(1, user_max_charge_current)
```

Comment:
There's a verification step in the test plan here, but I'm not sure what it means - it says "Charging rate reduced to 6A". Is that step 6a? Do you need to check that? Seems like it might be checked below. Is that just test plan weirdness?

Reply:
That check is done on line 224-225 below.
It is 6Amps. but the algorithm is based on the expected_max_charge = min(user_max_charge_current, circuit_capacity)